### PR TITLE
refactor: refactor api to utility classes

### PR DESCRIPTION
**Pull request by lukicenturi**
Part of #4872

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/store/defi/sushiswap/index.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const useSushiswapStore = defineStore('defi/sushiswap', () => {
   const isPremium = usePremium();
   const { activeModules } = useModules();
   const { t } = useI18n();
+  const api = useSushiswapApi();
```
Comment (kelsos) [inline frontend/app/src/store/defi/sushiswap/index.ts:31]: destructure

File: frontend/app/src/store/defi/yearn/index.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ export const useYearnStore = defineStore('defi/yearn', () => {
   const { assetSymbol } = useAssetInfoRetrieval();
   const premium = usePremium();
   const { t } = useI18n();
+  const api = useYearnApi();
```
Comment (kelsos) [inline frontend/app/src/store/defi/yearn/index.ts:37]: destructure

File: frontend/app/src/store/defi/index.ts [hunk 3]
```diff
@@ -78,6 +78,8 @@ export const useDefiStore = defineStore('defi', () => {
   const lendingStore = useDefiSupportedProtocolsStore();
   const { t, tc } = useI18n();
 
+  const api = useDefiApi();
```
Comment (kelsos) [inline frontend/app/src/store/defi/index.ts:81]: destructure

File: frontend/app/src/store/defi/uniswap.ts [hunk 4]
```diff
@@ -31,6 +31,8 @@ export const useUniswapStore = defineStore('defi/uniswap', () => {
   const isPremium = usePremium();
   const { t, tc } = useI18n();
 
+  const api = useUniswapApi();
```
Comment (kelsos) [inline frontend/app/src/store/defi/uniswap.ts:34]: destructure

File: frontend/app/src/store/history/asset-movements.ts [hunk 5]
```diff
@@ -43,6 +43,9 @@ export const useAssetMovements = defineStore('history/assetMovements', () => {
   const { fetchAssociatedLocations } = locationsStore;
   const { exchangeName } = useTradeLocations();
   const { tc } = useI18n();
+  const { notify } = useNotificationsStore();
+
+  const api = useAssetMovementsApi();
```
Comment (kelsos) [inline frontend/app/src/store/history/asset-movements.ts:48]: destructure

File: frontend/app/src/store/history/ledger-actions.ts [hunk 6]
```diff
@@ -42,6 +42,9 @@ export const useLedgerActions = defineStore('history/ledgerActions', () => {
   const { fetchAssociatedLocations } = useAssociatedLocationsStore();
   const { exchangeName } = useTradeLocations();
   const { tc } = useI18n();
+  const { notify } = useNotificationsStore();
+
+  const api = useLedgerActionsApi();
```
Comment (kelsos) [inline frontend/app/src/store/history/ledger-actions.ts:47]: destructure

File: frontend/app/src/store/history/trades.ts [hunk 7]
```diff
@@ -44,6 +44,9 @@ export const useTrades = defineStore('history/trades', () => {
   const { fetchAssociatedLocations } = locationsStore;
   const { exchangeName } = useTradeLocations();
   const { tc } = useI18n();
+  const { notify } = useNotificationsStore();
+
+  const api = useTradesApi();
```
Comment (kelsos) [inline frontend/app/src/store/history/trades.ts:49]: destructure

File: frontend/app/src/store/history/transactions.ts [hunk 8]
```diff
@@ -43,10 +45,15 @@ export const useTransactions = defineStore('history/transactions', () => {
   const counterparties = ref<string[]>([]);
 
   const { t } = useI18n();
+  const { notify } = useNotificationsStore();
+
+  const { ethAddresses } = storeToRefs(useEthAccountsStore());
+  const api = useTransactionsApi();
```
Comment (kelsos) [inline frontend/app/src/store/history/transactions.ts:51]: destructure

File: frontend/app/src/store/reports/index.ts [hunk 9]
```diff
@@ -99,10 +100,13 @@ export const useReports = defineStore('reports', () => {
   const { itemsPerPage } = storeToRefs(useFrontendSettingsStore());
   const { t } = useI18n();
 
+  const api = useReportsApi();
+  const historyApi = useHistoryApi();
```
Comment (kelsos) [inline frontend/app/src/store/reports/index.ts:104]: destructure

**Review by kelsos — CHANGES_REQUESTED** [review]
Thanks for taking this over.

My comments are mostly 

- about treating apis as composables, and destructuring them when using them.
- about converting `promise.then` to `async` `await` in the apis we touch in this pr
- about converting `function` to `const x => {}` in the apis we touch in this PR,

finally I also have a question to a change that doesn't seem related to the PR

File: frontend/app/src/components/settings/data-security/DataManagement.vue [hunk 10]
```diff
@@ -62,24 +64,28 @@ const showConfirmation = (source: PurgeParams) => {
 };
 
 const purgeSource = async (source: Purgeable) => {
+  const { deleteModuleData } = useBlockchainBalanceApi();
+  const { deleteEthTransactions } = useTransactionsApi();
+  const { deleteExchangeData } = useExchangeApi();
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply frontend/app/src/components/settings/data-security/DataManagement.vue:69]: sorry I don't get it, which setup context?

File: frontend/app/src/store/blockchain/index.ts [hunk 11]
```diff
@@ -164,19 +164,27 @@ export const useBlockchainStore = defineStore('blockchain', () => {
...
-          await enableModule({
-            enable: modules,
-            addresses: registeredAddresses
-          });
+          startPromise(
+            enableModule({
+              enable: modules,
+              addresses: registeredAddresses
+            })
+          );
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply frontend/app/src/store/blockchain/index.ts:174]: I change it to `startPromise` instead, so this function is done in the line before `await fetchAccounts(blockchain);`
So the bottom drawer can be closed immediately. (dont need to wait for the balances to be fetched first)

File: frontend/app/src/store/blockchain/index.ts [hunk 12]
```diff
@@ -164,19 +164,27 @@ export const useBlockchainStore = defineStore('blockchain', () => {
...
+        startPromise(
+          (async () => {
+            if (blockchain === Blockchain.ETH) {
+              await fetchDetected(registeredAddresses);
+            }
+            await refreshAccounts(blockchain);
+          })()
+        );
+
+        startPromise(fetchNonFungibleBalances(true));
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply frontend/app/src/store/blockchain/index.ts:187]: same like above, but for this make sure they are executed in order. (Is there any better way to write this)

File: frontend/app/src/store/blockchain/index.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (kelsos) [reply frontend/app/src/store/blockchain/index.ts:174]: I will merge as it is and think of a way. But if something needs to run sequentially we have to await, `startPromise` is for when we don't care about sequence, it is kind of fire and forget.

**Review by kelsos — APPROVED** [review]
I will do some small check changes on the blockchain account addition my self but the reset looks good.

**Pull request merged**